### PR TITLE
Add dcb (debug continue back) command

### PR DESCRIPTION
**Pull request by RKX1209**
Add new command 'dcb' that can continue backward until hit breakpoint.
When executing 'dcb', at first, get the previous breakpoint, and then, continue until this point.

### CODE REVIEW

File: libr/debug/debug.c [hunk 1]
```diff
@@ -1219,6 +1219,70 @@ R_API int r_debug_continue_until_nonblock(RDebug *dbg, ut64 addr) {
...
 
+R_API bool r_debug_continue_back(RDebug *dbg) {
+	RDebugSession *before;
+	RBreakpointItem *prev = NULL;
+	int has_bp;
+	ut64 pc, end_addr;
+
+	if (r_debug_is_dead (dbg))
+		return false;
+	if (!dbg->anal || !dbg->reg) {
```
Comment (XVilka) [inline libr/debug/debug.c]: Probably this should be before `r_debug_is_dead()` and also check `dbg` pointer too.

File: libr/debug/debug.c [hunk 2]
```diff
@@ -1219,6 +1219,70 @@ R_API int r_debug_continue_until_nonblock(RDebug *dbg, ut64 addr) {
 	return r_debug_continue_until_internal (dbg, addr, false);
 }
 
+R_API bool r_debug_continue_back(RDebug *dbg) {
+	RDebugSession *before;
+	RBreakpointItem *prev = NULL;
+	int has_bp;
+	ut64 pc, end_addr;
+
+	if (r_debug_is_dead (dbg))
```
Comment (radare) [inline libr/debug/debug.c]: Missing braces

File: libr/debug/debug.c [hunk 3]
```diff
@@ -1219,6 +1219,70 @@ R_API int r_debug_continue_until_nonblock(RDebug *dbg, ut64 addr) {
...
+	if (!has_bp)
+		r_bp_add_sw (dbg->bp, end_addr, dbg->bpsize, R_BP_PROT_EXEC);
+
+	// Continue until end_addr
+	for (;;) {
+		if (r_debug_is_dead (dbg))
+			break;
+		pc = r_debug_reg_get (dbg, dbg->reg->name[R_REG_NAME_PC]);
+		if (pc == end_addr)
+			break;
```
Comment (radare) [inline libr/debug/debug.c]: Missing braces

File: libr/debug/debug.c [hunk 4]
```diff
@@ -1219,6 +1219,70 @@ R_API int r_debug_continue_until_nonblock(RDebug *dbg, ut64 addr) {
...
+		return false;
+	}
+	//eprintf ("prev->addr = 0x%08"PFMT64x"\n", prev->addr);
+	/* Now we got previous breakpoint.
+	 * ### Continue until prev breakpoint ### */
+
+	/* Rollback to previous state again */
+	r_debug_session_set (dbg, before);
+	for (;;) {
+		if (r_debug_is_dead (dbg))
```
Comment (radare) [inline libr/debug/debug.c]: Braces

**Comment by radare** [comment]
As long as we are going back in ime maybe we should rename this isdead function to is_alive() to make more sense in the meaning that we can go back in time until the point of the process was not yet created.

Do dcb stops when there are no more traces?

**Comment by radare** [comment]
Can we use backstep in emulation with esil?

**Comment by radare** [comment]
doesnt builds on windows

**Comment by RKX1209** [comment]
still can't build on windows?

**Pull request merged**